queries/userQuery.ts

```diff
@@ -1,8 +1,20 @@
 import { gql } from '@apollo/client';
-import { UserResponse } from '@/models/user';
+import { User } from '@/models/user';
+import { WorkspaceSnapshot } from '@/models/workspace';
+import { Overwrite } from '@/utils/types';
 
 export interface UserQueryResponse {
-  user: UserResponse & { __typename?: 'User' };
+  user: Overwrite<
+    User,
+    {
+      workspaces: Pick<
+        WorkspaceSnapshot,
+        'id' | 'formCount' | 'timestamp' | 'users' | 'workspaceName'
+      >[];
+    }
+  > & {
+    __typename?: 'User';
+  };
```

Comment:
for example, if I had to guess you probably don't love the overwrite pick stuff.  My rationale for it is it lends some type safety to sticking closely with the backend model without directly coupling to it.  The alternative is to just make an interface that says:
```ts
export interface UserQueryResponse {
  id: string;
  workspaces: Array<{ id: string; formCount: number; timestamp: string; users: string[]; workspaceName: string; }>
  __typename?: 'User;
}
```

This is specifically a query response interface so it should not live with the backend user model. I tried to however bridge that gap using pick and overwrite. While we don't want to go overboard with that technique, it's a nice happy medium and more succint code a and adds some static safety if either of these interfaces change and could be split apart later.

Reply:
Yeah, I'm not big on it, it's not terrible. However, I don't think any of this is necessary anymore with all of the codegen I did in #20 . There won't be a need to type out the response, the hooks it exposes based on your schema will already have the associated types. Which is great, because it's always up to date with your real time schema, so you don't have to deal with the schema -> model maintenance anymore.